modules/private/internal.go

```
@@ -19,29 +20,10 @@ import (
 	"code.gitea.io/gitea/modules/setting"
 )
 
-func newRequest(ctx context.Context, url, method, sourceIP string) *httplib.Request {
-	if setting.InternalToken == "" {
-		log.Fatal(`The INTERNAL_TOKEN setting is missing from the configuration file: %q.
-Ensure you are running in the correct environment or set the correct configuration file with -c.`, setting.CustomConf)
-	}
-	return httplib.NewRequest(url, method).
-		SetContext(ctx).
-		Header("X-Real-IP", sourceIP).
-		Header("Authorization", fmt.Sprintf("Bearer %s", setting.InternalToken))
-}
-
-// Response internal request response
+// Response is used for internal request response (for user message and error message)
 type Response struct {
-	Err string `json:"err"`
-}
-
-func decodeJSONError(resp *http.Response) *Response {
-	var res Response
-	err := json.NewDecoder(resp.Body).Decode(&res)
-	if err != nil {
-		res.Err = err.Error()
-	}
-	return &res
+	Err     string `json:"err"`      // server-side error log message, it won't be exposed to end users
```

Erm…
How if you don't ignore it for the JSON deserialization?

---

Sorry, I haven't got your point.

---

I guess I might understand your question now.

* If you mean "how if affects the API client/caller", I think it's caller's duty to decide how to handle the error.

    For example, when we write code like ` f = OpenFile(...)`, then we can do  `err = f.Close()` to handle the error or `_ = f.Close()` to ignore the error. Let the caller decide.

* If you mean something else, I think I need a real example to understand the question.

---

Or do you mean "omitempty"? I will add (using it or not doesn't affect the logic)